server/src/Data/Data.hpp

```diff
@@ -0,0 +1,11 @@
+#ifndef DATA_HPP
+#define DATA_HPP
+
+struct Data
+{
+    //some data from application
+};
+
+using Callback = std::function<void(std::string, Data)>;
```

Comment:
Not Callback. Give some appropriate name, like `ProcessDataCallback` or `PrepareResponseCallback`